sql-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/BufferSendState.scala

```diff
@@ -152,7 +152,7 @@ class BufferSendState(
             requestHandler.acquireShuffleBuffer(bufferId)) { rapidsBuffer =>
             //these are closed later, after we synchronize streams
             rapidsBuffer.storageTier match {
-              case StorageTier.DEVICE =>
+              case StorageTier.DEVICE | StorageTier.GDS =>
```

Comment:
I think instead of doing this, we should have `BufferSendState` have a special case for GDS buffers, then have it call `CuFile.copyFileToDeviceBuffer` with a given offset (indirectly). That way we remove an extra copy and we don't need to allocate the full GDS buffer, in an already constrained device.

If you have a large GDS file this would allow you to, on the read path, to chunk those reads at most bounce buffer length also. Not sure if this has implications to GDS (several read requests, vs one large one, and what should the ideal request size be)

Reply:
I think that for an initial implementation it may be fine the way it is, but we may OOM more often, in the disk spill case since now we need more GPU memory in the interim.

Reply:
Yes probably need more tuning down the road.